src/graphql/schema/Trip/resolvers.js

```diff
@@ -0,0 +1,86 @@
+import {
+  users,
+  voters,
+  creator,
+  buildSuggestionsObj,
+  findUserOrCreate
+} from '../resolver-helpers';
+import { AuthenticationError } from 'apollo-server';
+
+export default {
+  Query: {
+    trip: (_, { id }, { Trip }) => Trip.findOne({ _id: id }),
+    allTrips: (_, __, { Trip }) => Trip.find()
+  },
+
+  Mutation: {
+    createTrip: async (_, { trip }, { Trip, User, user: { email } }) => {
+      const user = await User.findOne({ email });
+      if (!user) throw new AuthenticationError();
+      const {
+        destination = { isDictated: false },
+        budget = { isDictated: false },
+        timeFrame = { isDictated: false },
+        participants
+      } = trip;
+
+      trip.participants = await findUserOrCreate(participants, User);
+      trip.participants = [user._id, ...trip.participants];
```

Comment:
Why do you assign trip.participants twice with different contents? Do we need line 27 if it is not used?

Reply:
One assignment builds on the other. the second assignment basically adds the user that creates the trip into the participants array. could be put into one line but this way it is (admittedly only a little bit) more explicit.